R/update_people.R

```diff
@@ -0,0 +1,232 @@
+
+## We handle dependencies here (there's most likely better ways to do this but
+## whatever)
+
+if (!require(googlesheets4)) {
+  install.packages("googlesheets4")
+}
+
+if (!require(dplyr)) {
+  install.packages("dplyr")
+}
+
+if (!require(stringi)) {
+  install.packages("stringi")
+}
+
+
+## Some functions for capitalization of names
+### Atomic version
+simple_cap <- function(x) {
+  s <- strsplit(x, " ")[[1]]
+  paste(toupper(substring(s, 1, 1)), tolower(substring(s, 2)),
+        sep = "", collapse = " ")
+}
+
+### vectorised version
+capitalize <- function(x) unname(vapply(x, simple_cap, character(1)))
+
+## Function to enforce ascii characters
+enforce_ascii <- function(x) {
+  x <- as.character(x)
+  transformation  <- "Any-Latin; Latin-ASCII"
+  stringi::stri_trans_general(x, id = transformation)
+}
+
+
+
+
+
+#' Import member data from registration spreadsheet
+#'
+#' This function imports data on members of RECON from the google
+#' spreadsheet. It returns a `tibble`, keeping only non-rejected members.
+
+import_memberships <- function() {
+  url <- "https://docs.google.com/spreadsheets/d/16sEx6mPoFpmL53j2DXP1SXyzztaYdGlVh29VYS6ahRU/edit#gid=983075176"
+  out <- googlesheets4::read_sheet(url)
+  accepted <- c("regular member", "contributing member")
+  dplyr::filter(out, `Board decision` %in% accepted)
+}
+
+
+
+
+
+#' Generate yaml entries for one member
+#'
+#' This function will generate a yaml entry compatible with the people.md file
+#' for a single member. Its argument is a single-row tibble matching the
+#' structure of the output of `import_memberships`. The output is a character
+#' vector which can be fed to `cat()` when creating the new `people.md` file.
+#'
+#' @param x one line of membership tibble
+#'
+#' @param add_missing_pic a logical indicating if a default 'anonymous' pic
+#'   should be created to replace missing photos
+
+make_member_yaml <- function(x, add_missing_pic = FALSE) {
+
+  # Some columns are renamed for convenience
+  x <- dplyr::rename(x,
+                     position = "Position and expertise (short)",
+                     twitter = "Twitter account (full address)",
+                     github = "Github account (full address)",
+                     website = "Website")
+
+  
+  # Output is built incrementally
+  ## name
+  out <- paste("  - name:", x["First name"], x["Last name"], sep = " ")
+
+  ## image
+  img_txt <- tolower(paste0("    img: /img/people/",
+                            gsub(" ", "-", x["First name"]),
+                            "-",
+                            gsub(" ", "-",  x["Last name"]),
+                            ".jpg"))
+  img_txt <- gsub("&ouml;", "o", img_txt)
+
+  out <- c(out, img_txt)
+
+  ## if image does not exist, copy the anonymous pic by default
+  path_to_pic <- sub("^[ ]*img: ", "", img_txt)
+
+  ## handle non ascii characters like ö
+  path_to_pic <- gsub("&ouml;", "o", path_to_pic)
+
+  if (!file.exists(path_to_pic)) {
```

Comment:
@thibautjombart I think you want
``` 
!if (file.exists(paste0("..", path_to_pic)))
```

Reply:
Oh, and that line 96 was an ad hoc fix which is no longer needed. Thanks

Reply:
yep - wasn't quite sure as saw you were using stringi for the ascii but figured it wouldn't hurt :)